### PR TITLE
Fix hardlink handling

### PR DESCRIPTION
**Pull request by chrisburr**
I think this fixes the issue reported in: https://github.com/conda-forge/pypy3.6-feedstock/issues/83

### CODE REVIEW

File: recipe/activate-cross-python.sh [hunk 1]
```diff
@@ -26,7 +26,9 @@ if [[ "${CONDA_BUILD:-0}" == "1" && "${CONDA_BUILD_STATE}" != "TEST" ]]; then
     cp $sysconfigdata_fn $BUILD_PREFIX/venv/lib/$(basename $sysconfigdata_fn)
 
     # For recipes using {{ PYTHON }}
-    cp $BUILD_PREFIX/venv/cross/bin/python $PREFIX/bin/python
+    # Remove the file first as it might be a hardlink and make sure to resolve symlinks
+    rm $(readlink -f $PREFIX/bin/python)
```
Comment (joaander) [inline recipe/activate-cross-python.sh:30]: @chrisburr, I think this fix for linux aarch64 breaks osx arm64. I'm getting
```
Setting up cross-python
readlink: illegal option -- f
usage: readlink [-n] [file ...]
```
in the latest attempt to build gsd v2.5.3: https://github.com/conda-forge/gsd-feedstock/pull/51
[Here is the azure-pipelines output](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=526107&view=logs&j=8ca717d7-2f91-593f-2e86-7b14f42b0119&t=f56d028c-9d05-5be8-63d1-db69e7fd673f&l=281).

In attempts earlier this morning, osx-arm64 passed while linux-aarch64 failed. Now it is the other way around.

Comment (benmwebb) [reply recipe/activate-cross-python.sh:30]: FWIW, this is also breaking my osx-arm64 builds, e.g. conda-forge/imp-feedstock#15. I'm happy to try any fixes/workarounds you come up with.

Comment (h-vetinari) [reply recipe/activate-cross-python.sh:30]: Can confirm, also happening for numpy.

Comment (chrisburr) [reply recipe/activate-cross-python.sh:30]: Urgh...I checked this but apparently recent macOS versions now support this flag but older ones don't. Thanks for reporting, let's see if I can find a reasonable cross-platform alternative.

```bash
$ chrisburr$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.9.5
BuildVersion:	13F34
$ chrisburr$ readlink -h
readlink: illegal option -- h
usage: readlink [-n] [file ...]
```

```bash
$ sw_vers
ProductName:	macOS
ProductVersion:	12.4
BuildVersion:	21F79
$ readlink -h
readlink: illegal option -- h
usage: readlink [-fn] [file ...]
```

Comment (isuruf) [reply recipe/activate-cross-python.sh:30]: @chrisburr, do we really need to read the symlink? Can't we just do `rm $PREFIX/bin/python`?

Comment (chrisburr) [reply recipe/activate-cross-python.sh:30]: We do, without you end up deleting the symlink and leaving the `pypy3` orignal binary.

Comment (isuruf) [reply recipe/activate-cross-python.sh:30]: Ah, line 36's sed will interfere.
How about just doing,
```
rm $PREFIX/bin/python
rm $PREFIX/bin/python3
```

**Pull request merged**